etl/steps/data/garden/tourism/2023-05-05/unwto.py

```diff
@@ -0,0 +1,252 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+import numpy as np
+import pandas as pd
+from owid.catalog import Dataset, Table
+from structlog import get_logger
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    log.info("unwto.start")
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow: Dataset = paths.load_dependency("unwto")
+
+    # Read table from meadow dataset.
+    tb_meadow = ds_meadow["unwto"]
+
+    # Create a dataframe with data from the table.
+    df = pd.DataFrame(tb_meadow)
+
+    #
+    # Process data.
+    #
+    log.info("unwto.harmonize_countries")
+    df = geo.harmonize_countries(df=df, countries_file=paths.country_mapping_path)
+
+    # Set multi-level index and check that index is unique
+    df.set_index(["country", "year"], inplace=True)
+    assert df.index.is_unique, "Index is not well constructed"
+
+    # Convert all the columns of the DataFrame 'df' into float32 data type.
+    df = convert_columns_to_float32(df)
+
+    # Shorten the names of the columns in the DataFrame 'df'.
+    df = shorten_column_names(df)
+
+    # Calculate the sum of values by year in the DataFrame 'df' and store the result in 'df_sum'.
+    df_sum = calculate_sum_by_year(df)
+
+    # Reset the index of 'df_sum'.
+    df_sum.reset_index(inplace=True)
+
+    # Concatenate 'df' and 'df_sum' into a single DataFrame 'merged_df'.
+    merged_df = concatenate_dataframes(df, df_sum)
+
+    # Calculate the sum of values by year for the countries Bonaire, Sint Eustatius, and Saba
+    # in the DataFrame 'merged_df', and store the result in 'sum_bon_sint_saba'.
+    sum_bon_sint_saba = calculate_sum_by_year_Bonaire_Sint_Eustatius_Saba(merged_df)
+
+    # Drop rows in 'merged_df' that correspond to the countries 'Saba', 'Sint Eustatius',
+    # and 'Bonaire', and store the resulting DataFrame in 'merged_df_drop_'.
+    merged_df_drop_ = merged_df.drop(merged_df[merged_df["country"].isin(["Saba", "Sint Eustatius", "Bonaire"])].index)
+
+    # Concatenate 'merged_df_drop_' and 'sum_bon_sint_saba' into a single DataFrame 'merged_df_concat'.
+    # The rows of 'sum_bon_sint_saba' will be appended to 'merged_df_drop_'.
+    merged_df_concat = merged_df_drop_.append(sum_bon_sint_saba, ignore_index=True)
+
+    # Set index, check that it's unique and reset index
+    merged_df_concat = merged_df_concat.set_index(["country", "year"])
+    assert len(merged_df_concat.index.levels) == 2 and merged_df_concat.index.is_unique, "Index is not well constructed"
+    merged_df_concat = merged_df_concat.reset_index()
```

Comment:
You could use `assert not merged_df_concat[['country', 'year']].duplicated().any(), "Index is not well constructed"` instead.